velox/expression/ConjunctExpr.cpp

```diff
@@ -415,18 +416,89 @@ TypePtr ConjunctCallToSpecialForm::resolveType(
   return ConjunctExpr::resolveType(argTypes);
 }
 
+ExprPtr ConjunctCallToSpecialForm::optimizeAnd(
+    std::vector<ExprPtr>& compiledChildren) {
+  auto numChildren = compiledChildren.size();
+  for (vector_size_t i = numChildren - 1; i >= 0; i--) {
+    if (auto constantExpr = std::dynamic_pointer_cast<exec::ConstantExpr>(
+            compiledChildren[i])) {
+      if (!constantExpr->value()->isNullAt(0)) {
+        if (!constantExpr->value()->as<ConstantVector<bool>>()->valueAt(0)) {
+          return constantExpr;
+        } else {
+          if (i == 0 && compiledChildren.size() == 1) {
+            return constantExpr;
+          } else {
+            compiledChildren.erase(compiledChildren.begin() + i);
+          }
+        }
+      }
+    }
+  }
+
+  if (compiledChildren.size() == 1) {
+    return compiledChildren.front();
+  }
+  return nullptr;
+}
+
+ExprPtr ConjunctCallToSpecialForm::optimizeOr(
```

Comment:
This method is almost a copy-paste of the optimizeAnd method. The only difference is that AND looks for constant TRUE, while OR looks for constant FALSE. Let's refactor to reduce copy-paste.